### DISCUSSION TITLE
Buttons partially hidden on building enhancement window when using larger font theme

### DISCUSSION THREAD

**Discussion by andy5995** [widelands/widelands#5334] [General]
Related to https://github.com/widelands/widelands/issues/1294 and https://github.com/widelands/widelands/issues/2851

Using the 1.1~git48/98044ee@master/Debug version of WL and an add-on theme I recently uploaded through the add-ons menu, the Larger Font theme... No problems that I've seen 45 minutes into a game, except for the building enhancement window.

I was kinda wondering if there was something different about that window in the cpp source code that might account for it. If you think it's an easy fix, like matching up that code with how the other windows are created. I haven't looked at the code much yet... Most of my experience is with C not C++.

Anyway, if there won't be a fix for a while, I just thought I'd mention it in case it was important to note. If you think it might be an easy  or semi-easy fix and give me a little guidance to start, I'll see what I can do.

![image](https://user-images.githubusercontent.com/16764864/160343868-67148734-7b52-4c1f-95b6-f55006e9a0f2.png)

You can see the other windows are handling the font size ok.

![image](https://user-images.githubusercontent.com/16764864/160344366-b7a15fe7-3efe-49b6-913b-117b93bb7f9a.png)

![image](https://user-images.githubusercontent.com/16764864/160344472-3193bda0-f7f1-473f-bbc8-34c81323581e.png)

**Comment by Noordfrees** [comment]
The action confirmation window still uses the old-style layouting so it has a fixed size. It'll need to be converted to the new-style box layouting so it resizes itself to accommodate the size of its content.

Since the content itself is already using box layout this could be as simple as replacing the fixed-size layouting with a call `set_center_panel(main_box)` in the `ActionConfirm` constructor: 

https://github.com/widelands/widelands/blob/b21dc782d267bbf52f4f014e79022378904503b7/src/wui/actionconfirm.cc#L167-L169

However this would also mean that all variations of this window always resize themselves to their content and therefore all have different sizes. I am not sure if this desirable, or if it would be better to have a fixed horizontal and minimum vertical size with as-needed vertical size increase (which might be a bit harder).

**Reply by frankystone** [reply]
A problem might be that one has to add linebreaks manually if the width is not fixed?
